core/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -0,0 +1,301 @@
+    }
+
+    private TableRange findRowArea(Row row) {
+        int col1 = -1;
+        int col2 = -1;
+        for (Cell cell : row) {
+            Boolean blank = isBlank(cell);
+            if (col1 < 0 && Boolean.FALSE.equals(blank)) {
+                col1 = cell.getColumnIndex();
+                col2 = col1;
+            } else if (col1 >= 0 && col2 >= col1) {
+                if (Boolean.FALSE.equals(blank)) {
+                    col2 = cell.getColumnIndex();
+                } else if (Boolean.TRUE.equals(blank)) {
+                    break;
+                }
+            }
+        }
+        return col1 >= 0 && col2 >= col1 ? new TableRange(0, 0, col1, col2) : null;
+    }
+
+    private InputStream getInputStream(XlsxReadOptions options, byte[] bytes) throws FileNotFoundException {
+        if (bytes != null) {
+            return new ByteArrayInputStream(bytes);
+        }
+        if (options.inputStream() != null) {
+            return options.inputStream();
+        }
+        return new FileInputStream(options.file());
+    }
+
+    private Table createTable(Sheet sheet, TableRange tableArea, XlsxReadOptions options) {
+        // assume header row if all cells are of type String
+        Row row = sheet.getRow(tableArea.startRow);
+        List<String> headerNames = new ArrayList<>();
+        for (Cell cell : row) {
+            if (cell.getCellType() == CellType.STRING) {
+                headerNames.add(cell.getRichStringCellValue().getString());
+            } else {
+                break;
+            }
+        }
+        if (headerNames.size() == tableArea.endColumn - tableArea.startColumn + 1) {
+            tableArea.startRow++;
+        } else {
+            headerNames.clear();
+            for (int col = tableArea.startColumn; col <= tableArea.endColumn; col++) {
+                headerNames.add("col" + col);
+            }
+        }
+        Table table = Table.create(options.tableName());
+        List<Column<?>> columns = new ArrayList<>(Collections.nCopies(headerNames.size(), null));
+        for (int rowNum = tableArea.startRow; rowNum <= tableArea.endRow; rowNum++) {
+            row = sheet.getRow(rowNum);
+            for (int colNum = 0; colNum < headerNames.size(); colNum++) {
+                Cell cell = row.getCell(colNum + tableArea.startColumn, MissingCellPolicy.RETURN_BLANK_AS_NULL);
+                Column<?> column = columns.get(colNum);
+                if (cell != null) {
+                    if (column == null) {
+                        column = createColumn(headerNames.get(colNum), cell);
+                        columns.set(colNum, column);
+                        while (column.size() < rowNum - tableArea.startRow) {
+                            column.appendMissing();
+                        }
+                    }
+                    Column<?> altColumn = appendValue(column, cell);
+                    if (altColumn != null && altColumn != column) {
```

Comment:
Should this be if (altColumn != null && ! altColumn.equals(column))?

Reply:
No. If **appendValue** return a new **Column** instance, that **equals** the old, it should still replace it.